kotest-assertions/src/jvmMain/kotlin/io/kotest/assertions/jvmfailures.kt

```diff
@@ -24,19 +29,87 @@ object Failures {
       System.getProperty("kotest.failures.stacktrace.clean", "true") == "true"
 }
 
-actual fun cleanStackTrace(throwable: Throwable): Throwable {
+actual fun <T : Throwable> cleanStackTrace(throwable: T): T {
    if (Failures.shouldRemoveKotestElementsFromStacktrace)
       throwable.stackTrace = UserStackTraceConverter.getUserStacktrace(throwable.stackTrace)
    return throwable
 }
 
 /**
  * Creates an [AssertionError] from the given message. If the platform supports nested exceptions, the cause
- * is set to the given [cause]. If the platform supports stack traces, then the stack is cleaned of `io.kotest`
- * lines.
+ * is set to the given [cause].
+ */
+actual fun createAssertionError(message: String, cause: Throwable?): AssertionError = AssertionError(message, cause)
```

Comment:
We should use a null message when constructing junit4/5 assertions. The exception already adds the `expected: x but was: y` text, so passing in our message will cause it to be duplicated.

https://github.com/junit-team/junit4/blob/435d41f0d45cfdbc1a38e1ad4eb1d5300da533f9/src/main/java/org/junit/Assert.java#L838-L853

https://github.com/junit-team/junit5/blob/master/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java#L111-L123

Reply:
These are helper classes though right, and we're instantiating the exceptions directly.

Reply:
You're right, my apologies.